### PR TITLE
Removing duplicated kratos components?

### LINKED ISSUE #3852: [Trilinos][MPI] Compilation failing
After recent changes, the master branch does not seem to compile without errors any more. Dear @jcotela, @roigcarlo told me that you were the expert to turn to in this respect. Unfortunately, I do not understand the code well enough to suggest a correction, here. The error was first noticed in the afternoon of Jan16, 2019. The following error message is displayed:

[ 97%] Building CXX object applications/trilinos_application/CMakeFiles/KratosTrilinosApplication.dir/trilinos_application.cpp.o

In file included from /home/simon/Kratos_Main/applications/trilinos_application/trilinos_application.cpp:20:0:
/home/simon/Kratos_Main/applications/trilinos_application/custom_factories/trilinos_linear_solver_factory.h:123:102: error: redefinition of ‘Kratos::KratosComponents<TComponentType>::ComponentsContainerType Kratos::KratosComponents<TComponentType>::msComponents’
 typename KratosComponents<TComponentType>::ComponentsContainerType KratosComponents<TComponentType>::msComponents;
                                                                                                      ^
In file included from /home/simon/Kratos_Main/kratos/includes/kratos_application.h:23:0,
                 from /home/simon/Kratos_Main/applications/trilinos_application/trilinos_application.h:22,
                 from /home/simon/Kratos_Main/applications/trilinos_application/trilinos_application.cpp:19:
/home/simon/Kratos_Main/kratos/includes/kratos_components.h:445:68: note: ‘Kratos::KratosComponents<TComponentType>::ComponentsContainerType Kratos::KratosComponents<TComponentType>::msComponents’ previously declared here
 typename KratosComponents<TComponentType>::ComponentsContainerType KratosComponents<TComponentType>::msComponents;
                                                                    ^
applications/trilinos_application/CMakeFiles/KratosTrilinosApplication.dir/build.make:62: die Regel für Ziel „applications/trilinos_application/CMakeFiles/KratosTrilinosApplication.dir/trilinos_application.cpp.o“ scheiterte (english: the rule to generate ... failed)

It would be very kind, if you could have a look at the respective files, please. Thank you very much in advance.

### PR DESCRIPTION
**Pull request by roigcarlo**
I suppose this will fix #3852. Still not sure why it was duplicated

### CODE REVIEW

File: applications/trilinos_application/custom_factories/trilinos_linear_solver_factory.h [hunk 1]
```diff
@@ -119,8 +119,8 @@ typedef LinearSolverFactory<TrilinosSparseSpaceType,  TrilinosLocalSpaceType> Tr
     KratosComponents<TrilinosLinearSolverFactoryType>::Add(name, reference);
 
 
-template<class TComponentType>
-typename KratosComponents<TComponentType>::ComponentsContainerType KratosComponents<TComponentType>::msComponents;
+// template<class TComponentType>
+// typename KratosComponents<TComponentType>::ComponentsContainerType KratosComponents<TComponentType>::msComponents;
```
Comment (philbucher) [inline applications/trilinos_application/custom_factories/trilinos_linear_solver_factory.h]: could you please remove and not comment this?
I am usually getting confused when I see stuff like this :)

**Comment by jcotela** [comment]
This can be safely removed. Apparently it was added after I created the branch for the clang warning fixes, which is why I missed it. I agree with @philbucher that removing it would be better.

If we ever add a KRATOS_API to the class, I am afraid we will have to also add the conditional `extern` for windows (I guess we are lucky we do not support MPI in windows...)

File: applications/trilinos_application/custom_factories/trilinos_linear_solver_factory.h (hunk 1 above)
Comment (roigcarlo) [reply applications/trilinos_application/custom_factories/trilinos_linear_solver_factory.h]: Done!

**Review by philbucher — APPROVED** [review]
Thanks
Also works for me btw

**Review by jcotela — APPROVED** [review]
Thanks!

**Pull request merged**